From 1386249b5f674d35029793f62b27391adacb0d63 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Tue, 4 Apr 2023 11:25:56 +0200
Subject: [PATCH 4/6] SSDK: init: remove gpio_set_debounce()

gpio_set_debounce() was dropped upstream, so lets easily convert to
gpiod_set_debounce().

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 src/init/ssdk_init.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/init/ssdk_init.c b/src/init/ssdk_init.c
index 84db1440..f35e53e2 100755
--- a/src/init/ssdk_init.c
+++ b/src/init/ssdk_init.c
@@ -51,6 +51,7 @@
 #include <linux/string.h>
 #include <linux/gpio.h>
 #include <linux/of_gpio.h>
+#include <linux/gpio/consumer.h>
 
 #if defined(IN_SWCONFIG)
 #if defined(CONFIG_OF) && (LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0))
@@ -1451,7 +1452,7 @@ static int config_gpio(a_uint32_t  gpio_num)
 			SSDK_ERROR("gpio request faild \n");
 			return -1;
 		}
-		gpio_set_debounce(gpio_num, 60000);
+		gpiod_set_debounce(gpio_to_desc(gpio_num), 60000);
 	}
 	else
 	{
-- 
2.41.0

